### DISCUSSION TITLE
ResponsiveValue from 'styled-system' doesn't work

### DISCUSSION THREAD

**Discussion by tomtobac** [twilio-labs/paste#659] [Q&A] [answered]
In Typescript we are not getting the autocomplete of the space because `ResponsiveValue` utility returns any...

> node_modules/@twilio-paste/style-props/dist/types/space.d.ts
```ts
import { ThemeShape } from '@twilio-paste/theme';
import { ResponsiveValue } from '@twilio-paste/styling-library';
export declare type SpaceOptions = keyof ThemeShape['space'];
export declare type Space = ResponsiveValue<SpaceOptions>;
export declare type Padding = Space;
export declare type Margin = Space | 'auto';
export interface MarginProps {
    margin?: Margin;
    marginTop?: Margin;
    marginRight?: Margin;
    marginBottom?: Margin;
    marginLeft?: Margin;
}
export interface PaddingProps {
    padding?: Padding;
    paddingTop?: Padding;
    paddingRight?: Padding;
    paddingBottom?: Padding;
    paddingLeft?: Padding;
}
export interface SpaceProps extends MarginProps, PaddingProps {
}
//# sourceMappingURL=space.d.ts.m
```

Margin
![image](https://user-images.githubusercontent.com/10562610/90271797-aaac5680-de5c-11ea-9e61-432e7b84f49a.png)

Space
![image](https://user-images.githubusercontent.com/10562610/90271762-9a947700-de5c-11ea-8126-c20eb3a9476c.png)

Space options
![image](https://user-images.githubusercontent.com/10562610/90271726-8badc480-de5c-11ea-8dfb-c82e5c68c22e.png)

**Comment by tomtobac** [comment, marked as answer]
I couldn't find this utility in `styled-system`... even the the package `@twilio-paste/styling-library` doesn't export this method... 
![image](https://user-images.githubusercontent.com/10562610/90272201-5ce41e00-de5d-11ea-8a20-5d79e17d06f6.png)

**Reply by TheSisb** [reply]
The `@twilio-paste/styling-library` package has two `devDependencies` that you might need. Try adding 
```
    "@types/styled-system": "^5.1.9",
    "@types/styled-system__css": "^5.0.5"
```
to your project and see if that fixes the issue.  If that fixes it, I'll move these as `dependencies` of the package since they're just typescript definitions so it shouldn't cause any issues.

**Reply by tomtobac** [reply]
hey @TheSisb, installing `@types/styled-system` fixes the issue. Thanks!

**Reply by TheSisb** [reply]
Awesome! I made a PR to fix this https://github.com/twilio-labs/paste/pull/663

**Comment by corinne-nagel** [comment]
Hi @tomtobac I've reached out to the team and should have a response for you today.

**Comment by richbachman** [comment]
Hi @tomtobac, will you share a code sample showing where you're getting the error for `space`? Any information about the your project environment would be helpful as well. Thanks.

**Reply by tomtobac** [reply]
hey @richbachman, it's our internal role [sip-connectivity](https://code.hq.twilio.com/voice-n-video-console/vnv-console-sip-connectivity)
